labgrid/resource/udev.py

```diff
@@ -289,29 +289,48 @@ class USBSDMuxDevice(USBResource):
     """The USBSDMuxDevice describes an attached USBSDMux device,
     it is identified via USB using udev
     """
+
+    # Explanation: The USBSDMux consists of two devices, one block and one scsi
+    # generic device. The manager will mark the resource available as soon as
+    # the parent device is added. But the SDMux has not been fully initialized
+    # yet. Thats why this Device overwrites the avail attribute with an
+    # internal avail.
+    # The internal avail has to be an attribute, since attrs can only resolve
+    # its attributes during validation.
+    internal_avail = attr.ib(default=False)
+
     def __attrs_post_init__(self):
         self.match['ID_VENDOR_ID'] = '0424'
         self.match['ID_MODEL_ID'] = '4041'
         self.control_path = None
         self.disk_path = None
         super().__attrs_post_init__()
 
-    def update(self):
-        block_found = False
+    # Overwrite the avail attribute with our internal property
+    @property
+    def avail(self):
+        return self.internal_avail
 
-        super().update()
+    # Forbid the USBResource super class to set the avail property
+    @avail.setter
+    def avail(self, prop):
+        pass
+
+    # Overwrite the poll function. Only mark the SDMux as available if both
+    # paths are available.
+    def poll(self):
+        super().poll()
         if not self.device:
             self.control_path = None
             self.disk_path = None
-            return
-        for child in self.device.children:
-            if child.subsystem == 'block':
-                self.disk_path = child.device_node
-                block_found = True
-            if child.subsystem == 'scsi_generic':
-                self.control_path = child.device_node
-        if not block_found:
-            self.disk_path = None
+        else:
+            if not self.avail:
```

Comment:
Using the property insead of the attr here seems strange. `if not (self.disk_path and self.control_path)`?